MetaMask/metamask-extension
# Update tokens controller

## Explanation

Trigger the transaction popup using calls to the ApprovalController instead of the showUserConfirmation callback.

Removes `openPopup` function as it's no longer being used.

See: https://github.com/MetaMask/MetaMask-planning/issues/410

## Manual Testing Steps

- Go to [dapp test page](https://metamask.github.io/test-dapp/)
- Connect wallet
- Click on "CREATE TOKEN"
- Click on "ADD TOKEN TO WALLET" to verify that the confirmation dialog works

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
